src/Features/CSharpTest/ExtractMethod/ExtractLocalFunctionTests.cs

```diff
@@ -5802,4 +5802,46 @@ public C(int x, System.Func<int, int> modX)
 
         await TestAsync(code, expected, TestOptions.Regular, index: CodeActionIndex);
     }
+
+    [Theory, WorkItem("https://github.com/dotnet/roslyn/issues/63917")]
+    [InlineData("true")]
+    [InlineData("false")]
+    public async Task TestAfterEndif(string ppDirective)
+    {
+        var code = $$"""
+            class C
+            {
+                void M()
+                {
+                    [|Console.WriteLine("test1");|]
+                    Console.WriteLine("test2");
+
+            #if {{ppDirective}}
+                    Console.WriteLine("test3");
+            #endif
+                }
+            }
+            """;
+
+        var expected = $$"""
+            class C
+            {
+                void M()
+                {
+                    {|Rename:NewMethod|}();
+                    Console.WriteLine("test2");
+            
+            #if {{ppDirective}}
+                    Console.WriteLine("test3");
+            #endif
+                    static void NewMethod()
```

Comment:
>static void NewMethod() [](http://example.com/codeflow?start=20&length=23)

should there be another newline between the #endif and this line?

Reply:
i didn't want to  get ultra fiddly here.  That's more the domain of improvements to the formatter.